### PR TITLE
Add pangram exercise

### PR DESCRIPTION
**Pull request by palisaide**
This will probably need to be updated depending on the outcome of #52 and #53

### CODE REVIEW

File: exercises/pangram/src/example.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#include "pangram.h"
+#include <string.h>
+
+#define NUMBER_OF_LETTERS (26)
+#define DOWNCASE(letter) ((letter >= 'A') && (letter <= 'Z')) ? letter + ('a' - 'A') : letter
+
+bool is_pangram(const char *sentence) {
+   int sentence_length = strlen(sentence);
```
Comment (ryanplusplus) [inline exercises/pangram/src/example.c]: `strlen` actually returns a `size_t` so you should use that here and for `i`.

File: exercises/pangram/src/example.c [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+   char found_letters[NUMBER_OF_LETTERS];
+   int i;
+
+   memset(found_letters, 0, sizeof(found_letters));
+
+   for(i = 0; i < sentence_length; i++) {
+      char letter = DOWNCASE(sentence[i]);
+
+      if((letter >= 'a') && letter <= 'z') {
```
Comment (ryanplusplus) [inline exercises/pangram/src/example.c]: Inconsistent use of parens (included on the first check, omitted on the second).

File: exercises/pangram/src/example.h [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#ifndef _PANGRAM_H
+#define _PANGRAM_H
+
+#include <stdbool.h>
+
+bool is_pangram(const char *sentence);
+#endif
```
Comment (ryanplusplus) [inline exercises/pangram/src/example.h]: Can you add a line of whitespace above to be consistent with the formatting of other exercises?

File: config.json [hunk 4]
```diff
@@ -74,6 +75,15 @@
       "bitwise operations",
       "performance optimizations"
     ]
+  }, {
+    "difficulty": 3,
+    "slug": "pangram",
+    "topics": [
+      "filtering",
```
Comment (ryanplusplus) [inline config.json]: I'm not sure that this really covers filtering. I think this really refers to whittling down a collection to a smaller collection.

**Comment by ryanplusplus** [comment]
Looks good overall, thanks

**Comment by patricksjackson** [comment]
Apparently @ryanplusplus and I were going through this at the same time, there may be a bit of overlap.

On the testing:

1. Please add more of the test cases from the [common json](https://github.com/exercism/x-common/blob/master/pangram.json), at the very least use one with numbers to catch people accidentally testing for `isalnum()`.
2. For C we should also have a NULL test case.

Now the example solutions shouldn't need to be overly fancy, just something that passes all the tests, but it does need a couple changes:

1. `size_t i, sentence_length;`
2. Your solution would fail if an input was given that overflowed `char`.
3. Using `isalpha()` and `tolower()` may be better to introduce people to the library functions.

Actually, in all honesty I've never been clear on what requirements we should actually place on the example solutions, but those seem like good things to have.

**Comment by ryanplusplus** [comment]
@patricksjackson good catches. I definitely agree on `tolower` and `isalpha`. I am fortunate to not have to do much text processing in C so I didn't even think of these.

As far as the requirements for the example solutions, I am of the mind that the sample solutions should be something that someone could learn from if they were to stumble upon this repo in search of better ways to complete the exercises. I strive to write "refrigerator code" (code you'd take home and put on your parents' refrigerator) for sample solutions, so I would encourage you to keep the nitpicks coming!

File: exercises/pangram/src/example.c [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+         found_letters[letter - 'a']++;
```
Comment (ryanplusplus) [inline exercises/pangram/src/example.c]: How about using a `bool` array and changing this to:

```
found_letters[letter - 'a'] |= true;
```

Comment (patricksjackson) [inline exercises/pangram/src/example.c]: That would've been my suggestion as well (though you only need `=` and not `|=`)

Comment (ryanplusplus) [inline exercises/pangram/src/example.c]: Oh derp. Good call.

**Comment by palisaide** [comment]
Thanks for the comments!  So, I totally forgot that there was a common json for most of these exercises; I was going through other languages trying to find test cases.

I agree with all suggestions.

**Comment by ryanplusplus** [comment]
Changes look good to me. I'm going to give @patricksjackson a chance to review them and then I'll merge.

**Comment by patricksjackson** [comment]
LGTM :+1:

**Pull request merged**